### DISCUSSION TITLE
How does super_fetch execute the rescued jobs?

### DISCUSSION THREAD

**Discussion by ShabiniRajadas** [sidekiq/sidekiq#5046] [General]
So we recently introduced super_fetch in our rails app. We could see the jobs are rescued and moved to corresponding queues when there is a sidekiq shutdown. Now when these orphaned jobs are executed again, do they start from the beginning of the worker, or do they start back from the place they were interrupted?

**Comment by mperham** [comment]
Sidekiq just calls `perform`. You must make your jobs idempotent: https://github.com/mperham/sidekiq/wiki/Best-Practices#2-make-your-job-idempotent-and-transactional
